experimentation/src/main/java/com/automattic/android/experimentation/local/FileBasedCache.kt

```diff
@@ -54,6 +47,8 @@ internal class FileBasedCache(
 
     suspend fun saveAssignments(assignments: Assignments) {
         withContext(dispatcher) {
+            assignmentsFile.parentFile?.mkdirs()
```

Comment:
It is probably my personal preference but I think it makes more sense to use `!!.` operator instead of `?.`. If the `parentFile` is `null` it is a code error in the library since we know what `assignmentsFile` is exactly. Alternatively it might be `requireNotNull(assignmentsFile.parentFile) { "error message" }` to provide more context.

Reply:
In this specific case, the crash will happen anyway because `file.writeText()` is called later in `saveAssignments()`. If the file doesn’t exist, this method will fail. Having the crash occur earlier would indicate the problem at a more appropriate point in the code.

As I mentioned before, this is likely my personal design approach. `parentFile` must exist here. We know this because it's defined as part of `assignmentsFile`, and it’s correct for the code to crash if `parentFile` doesn’t exist. Ignoring this condition leads to unpredictability in the library. If the code neither crashes nor returns an error, the consumer might remain unaware of the issue for a long time. Although in this case, the issue will be detected quickly due to the `file.writeText()` call.

I'm not expecting a change here but I wanted to offer my 2 cents about the library design. Having it logged doesn't change much in my opinion but it's better than nothing. It's also worth mentioning that the code is covered with a test so it's more of a philosophical than practical point.